### PR TITLE
fix(ci): Fix pattern so the Performance Benchmarks pipeline doesn't choke on packages with overlapping names

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Fixes a "glob" pattern (ADO-specific syntax) so the Performance Benchmarks pipeline can differentiate the `tgz` files for packages that have partially overlapping names (e.g. `@fluid-experimental/tree` and `@fluid-experimental/tree-react-api`).

Follow-up to https://github.com/microsoft/FluidFramework/pull/22995.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

[AB#17128](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/17128)

### CODE REVIEW

File: tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml [hunk 1]
```diff
@@ -41,7 +41,16 @@ steps:
...
+
+      # There's some logic behind the value for the itpbip_sanitizedPackageName.
+      # We've had issues in the past where two packages with partially overlapping names (e.g. @fluid-experimental/tree
+      # and @fluid-experimental/tree-react-api) both match a simpler pattern and then cause issues later on when
+      # trying to unpack. Adding a dash and a number after the package name should be enough to avoid this issue.
+      # It also solves a problem with another pattern we had used (?.?.?-*.tgz), which would break for a two digit
+      # major/minor/patch version numbers.
+      # The fact that we want at least one number after the package name should be a good enough heuristic, since we
+      # don't have any packages with a number in the name.
+      echo "##vso[task.setvariable variable=itpbip_sanitizedPackageName]${TEST_PACKAGE_NAME}-[0-9]*.tgz"
```
Comment (alexvy86) [inline tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml:53]: Note that this is not a regex-like "any number of characters (`*`) that match `0-9`", it means "one character that matches `0-9` followed by any number of characters (`*`) _of any kind_". [Doc reference](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/file-matching-patterns?view=azure-devops).

**Review by anthony-murphy — APPROVED** [review]

File: tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml [hunk 2]
```diff
@@ -41,7 +41,29 @@ steps:
...
+      # The logic behind the current pattern is that we want to find these components:
+      # - `${TEST_PACKAGE_NAME}` - Matching package name.
+      # - `-` A dash.
+      # - `+([0-9]).+([0-9]).+([0-9])` - The version (i.e. three sets of one or more digits separated by dots).
+      #   This component *must not* match another bit of package name. As long as we don't put numbers in package names,
+      #   the current pattern should keep working.
+      # - `*` - Potentially more characters (to handle the case where "-<build-id>" is appended to the version number).
+      # - `.tgz` - The extension.
+      # NOTE: this is not standard regex syntax, but ADO globbing.
+      # See https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/file-matching-patterns?view=azure-devops#pattern-syntax for details.
```
Comment (jason-ha) [inline tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml:65]: I thought there was a use case where the glob was used with `ls` command. At least I have a comment relating to `ls` in Teams discussion thread.

Comment (alexvy86) [reply tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml:67]: Thanks for the callout, I had forgotten about that and I think you're right. I'll confirm and update this.

**Comment by alexvy86** [comment]
This version of the pattern works. [Test run of Performance Benchmarks](https://dev.azure.com/fluidframework/internal/_build/results?buildId=304743&view=results) (msft internal). The failure is unrelated, a real-service test that happened to fail. I'll fix the conflict and go ahead merging this. Of note: this doesn't validate that the pattern works for pipeline runs made during a release because we'd need to actually do one (to validate the case where the tgz filenames have versions like `x.y.z.` and not `x.y.z-123456`). We'll know when 2.10.0 happens in a few weeks.

**Pull request merged**